kubernetes/template-generator/template_generator.py

```diff
@@ -0,0 +1,341 @@
+            in the same directory (scope) names as: sanitized-container-name.tar.gz
+
+            @return string  a dockerfile content
+        """
+        # TODO: This can be done better if the image is being build on the same machine
+        #       where the leapp tool is running
+        path = "{}.tar.gz".format(self.container_name)
+        content = "FROM {0}\nADD {1} /".format(self.system.base_image(), path)
+
+        return content
+
+
+class MultivolumeTemplateGenerator(TemplateGenerator):
+    FORBIDDEN_MOUNTS = ["dev", "sys", "proc", "lost+found"]
+
+    def __init__(self, container_name, system, image_url, exposed_ports=None, external_ips=None,
+                 exported_paths=None):
+        """
+            @param string container_name    the name of migrated container
+            @param ContainerOS system       a os description module
+            @param string image_url         an URL to tar image containing the container FS
+            @param list exported_paths      a list of all paths for which should be created a volume
+            @param list exposed_ports       a list of exposed ports
+        """
+        TemplateGenerator.__init__(self, container_name, system, exposed_ports, external_ips)
+
+        if exported_paths is None or len(exported_paths) == 0:
+            raise AttributeError("Exported paths has to be set")
+
+        if not len(image_url):
+            raise AttributeError("image_url cannot be empty")
+
+        self.image_url = image_url
+        self.exported_paths = list(exported_paths)
+
+        # Not allowed paths
+        for FORBIDDEN_MOUNT in self.FORBIDDEN_MOUNTS:
+            try:
+                self.exported_paths.remove(FORBIDDEN_MOUNT)
+            except Exception:
+                pass
+
+    def generate_pod_template(self, yaml=True):
+        """
+            Generate a YAML template for kerberos pod
+            @param Boolean yaml     if true, the output will be in yaml, else a
+                                    dict will be returned
+
+            @return string/dict    a template string in yaml format/dict
+        """
+        def _generate_common_mounts(prefix="", mounts=None):
+            """
+                Generate mounts for a container
+
+                @param string prefix    a prefix path for each generated mount
+                @param list mounts      a list of mounts (optional)
+
+                @return array           array of dicts describing mount points. If
+                                        mounts was given, the returned array will be
+                                        just a pointer to the same.
+            """
+            mounts = mounts or []
+
+            for exported_path in self.exported_paths:
+                vol_name = "{}-{}-vol".format(self.container_name, exported_path)
+
+                if not len([mount for mount in mounts if mount["name"] == vol_name]):
```

Comment:
This with the previous comment is better than iterating over all mounts every time